Lib/fontTools/ttLib/tables/_h_h_e_a.py

```diff
@@ -32,56 +34,63 @@ class table__h_h_e_a(DefaultTable.DefaultTable):
 
 	# Note: Keep in sync with table__v_h_e_a
 
-	dependencies = ['hmtx', 'glyf']
+	dependencies = ['hmtx', 'glyf', 'CFF ']
 
 	def decompile(self, data, ttFont):
 		sstruct.unpack(hheaFormat, data, self)
 
 	def compile(self, ttFont):
-		if ttFont.isLoaded('glyf') and ttFont.recalcBBoxes:
+		if ttFont.recalcBBoxes and (ttFont.isLoaded('glyf') or ttFont.isLoaded('CFF ')):
 			self.recalc(ttFont)
 		self.tableVersion = fi2ve(self.tableVersion)
 		return sstruct.pack(hheaFormat, self)
 
 	def recalc(self, ttFont):
-		hmtxTable = ttFont['hmtx']
+		boundsWidthDict = {}
 		if 'glyf' in ttFont:
 			glyfTable = ttFont['glyf']
-			INFINITY = 100000
-			advanceWidthMax = 0
-			minLeftSideBearing = +INFINITY  # arbitrary big number
-			minRightSideBearing = +INFINITY # arbitrary big number
-			xMaxExtent = -INFINITY          # arbitrary big negative number
-
 			for name in ttFont.getGlyphOrder():
-				width, lsb = hmtxTable[name]
-				advanceWidthMax = max(advanceWidthMax, width)
 				g = glyfTable[name]
 				if g.numberOfContours == 0:
 					continue
 				if g.numberOfContours < 0 and not hasattr(g, "xMax"):
 					# Composite glyph without extents set.
 					# Calculate those.
 					g.recalcBounds(glyfTable)
-				minLeftSideBearing = min(minLeftSideBearing, lsb)
-				rsb = width - lsb - (g.xMax - g.xMin)
-				minRightSideBearing = min(minRightSideBearing, rsb)
-				extent = lsb + (g.xMax - g.xMin)
-				xMaxExtent = max(xMaxExtent, extent)
+				boundsWidthDict[name] = g.xMax - g.xMin
+		elif 'CFF ' in ttFont:
+			topDict = ttFont['CFF '].cff.topDictIndex[0]
+			for name in ttFont.getGlyphOrder():
+				cs = topDict.CharStrings[name]
+				if cs.bounds is None:
+					continue
+				boundsWidthDict[name] = math.ceil(cs.bounds[2]) - math.floor(cs.bounds[0])
+
+		advanceWidthMax = 0
+		minLeftSideBearing = float('inf')
+		minRightSideBearing = float('inf')
+		xMaxExtent = -float('inf')
```

Comment:
Same here, those could be None then below in the for:

~~~
if xMaxExtent is None:
    minLeftSideBearing = lsb
    minRightSideBearing = rsb
    xMaxExtent = extent
else:
    minLeftSideBearing = max(minLeftSideBearing, lsb)
    minRightSideBearing = max(minRightSideBearing, rsb)
    xMaxExtent = max(xMaxExtent, extent)
~~~

I feel it's more pythonic this way.